### PR TITLE
Reorganize esil directory

### PR DESCRIPTION
**Pull request by Heersin**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

1. Move the esil-analysis codes to `analysis/esil` 
2. Move esil codes in `core/canalysis.c` to `core/cil.c`

**Test plan**

Should pass the old rizin tests.

### CODE REVIEW

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
@Heersin please run `clang-format`

**Comment by Heersin** [comment]
> @Heersin please run `clang-format`

Ok

File: librz/core/cil.c [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+	} else {
+		eprintf("Cannot deinitialize %s\n", stack_name);
+	}
+	rz_flag_unset_name(core->flags, stack_name);
+	rz_flag_unset_name(core->flags, "aeim.stack");
+	sdb_unset(core->sdb, "aeim.fd", 0);
+	free(stack_name);
+	return;
+}
+/**
```
Comment (XVilka) [inline librz/core/cil.c:246]: MIssing new line.

File: librz/core/cil.c [hunk 2]
```diff
@@ -0,0 +1,424 @@
+// SPDX-FileCopyrightText: 2009-2020 pancake <pancake@nopcode.org>
+// SPDX-FileCopyrightText: 2009-2020 nibble <nibble.ds@gmail.com>
+// SPDX-FileCopyrightText: 2021-2021 heersin <teablearcher@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
```
Comment (XVilka) [inline librz/core/cil.c:4]: Please add a new line here

**Review by XVilka — CHANGES_REQUESTED** [review]

**Review by wargio — APPROVED** [review]

File: librz/core/cil.c (hunk 2 above)
Comment (XVilka) [reply librz/core/cil.c:4]: Also please don't write `2021-2021`, just `2021`.

**Review by XVilka — APPROVED** [review]

**Comment by XVilka** [comment]
OpenBSD failure is unrelated.

**Review by ret2libc — APPROVED** [review]

**Pull request merged**